### PR TITLE
Add brakeman security scanner

### PR DESCRIPTION
**Pull request by bf4**
Run via `script/brakeman`

- Intentionally runs outside of bundle, per author recommendation, so that it always uses the latest version
- Uses brakeman-min since we don't need csv, haml, slim output
- Since there are currently warnings, I have set `exit_on_warn` to `false` so that the build can pass, and I think fixing the warnings is outside the scope of this PR.

Report diffing via caching (new for this repo): caching reports in the 'reports' directory. Per
http://docs.travis-ci.com/user/caching/#Enabling-multiple-caching-features (but this won't work on a public repo until we're back on 'our new container-based infrastructure'.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ gem 'aws-sdk-core'
 gem 'gchartrb', require: 'google_chart'
 gem 'gravtastic'
 gem 'high_voltage'
-gem 'highline'
+gem 'highline', '~> 1.6'
```
Comment (bf4) [inline Gemfile]: ref: https://github.com/presidentbeef/brakeman/pull/685

File: script/brakeman [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+# Does not install the latest brakeman if already installed
+gem install brakeman --conservative
+ruby -rbrakeman -S rake -f lib/tasks/brakeman.rake brakeman
```
Comment (bf4) [inline script/brakeman]: `ruby -rbrakeman -S` is to ensure the command is run without bundled gems. e.g. rubygems-bundler installed by rvm would otherwise cause this script to fail.

**Comment by sferik** [comment]
FYI, I just rebased this branch from `master` after fixing some issues there that were causing RuboCop to fail the build.

**Comment by bf4** [comment]
@sferik Anything you'd want to change before merging?  Do you like having this? Running it outside of the bundle (as the gem author recommends)?

File: config/brakeman.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+:min_confidence: 3
+:ignore_ifs: false
+:skip_libs: false
+:exit_on_warn: false
```
Comment (bf4) [inline config/brakeman.yml]: Since there are currently warnings, I have set `exit_on_warn` to false so that the build can pass, and I think fixing the warnings is outside the scope of this PR.

**Comment by bf4** [comment]
Any blockers on this? @sferik

File: Gemfile [hunk 4]
```diff
@@ -9,7 +9,7 @@ gem 'aws-sdk-core'
 gem 'gchartrb', require: 'google_chart'
 gem 'gravtastic'
 gem 'high_voltage'
-gem 'highline'
+gem 'highline', '~> 1.7'
```
Comment (sferik) [inline Gemfile]: Why is this necessary?

Comment (bf4) [inline Gemfile]: When I first tried to install brakeman in the Gemfile, it failed because brakeman wanted highline 1.6.x. So, I promoted the 1.7 version constraint of highline from the Gemfile.lock to the Gemfile.

I'm not familiar with how the rg tracks and updates dependent gems.

If your gut it to remove it, I will, no problem.
